### ISSUE TITLE
Game crash with AdventofAscension (Advancements crashing)

### ISSUE THREAD

**Issue by MudkipNinja** [Coders-After-Dark/Reskillable#163]
So I've created a modpack with Reskillable-1.12.2-1.12.5, and Advent of Ascension(AoA3-3.1.2.b). Whenever I have the two mods enabled together, the game crashes. I've attached the crash report, but after chatting the the developer of AoA, I've determined that its because of the way advancements are handled between the two mods. 

I've already asked the developer of AoA if he could change the way he handles advancements, but he said that it would be much harder and that the problem originates from this mod.

If this is an easy incompatibility to solve, then I'd love to know, otherwise, I'm ditching AoA.

[crash-2019-08-17_16.01.48-server.txt](https://github.com/Coders-After-Dark/Reskillable/files/3512409/crash-2019-08-17_16.01.48-server.txt)

**Comment by wootaboot** [comment]
I have come to the same issue. I image its going to be pushed off to a different version of the game so rip.

**Comment by MudkipNinja** [comment]
Still no update on this :(

**Comment by Cicopath** [comment]
Yeah, I'm having the same exact problem. I was wondering why I was able to create a New Server. Then, when I left and rejoined the server, I'd constantly crash when the world was loading. I was trying to figure this out with the creator of AoA as well, and he said it might be Reskillable. And they were right, it was Reskillable conflicting with AoA, nothing else. I even have the report from both Minecraft and AoA's crash report thing in the config file.

**Comment by Tslat** [comment]
It's not so much that it's _hard_ to fix it from my side, it's that to fix it, I basically have to remove any functionality involved in causing it.
It's _my_ advancements that are being messed with, I should be able to expect that an advancement I added and control would be in a state I expect it to be in.

If there's functionality reskillable has that requires it, it might be better off adding a config option like triumph does that stops it from removing other mod's advancements

**Comment by pupnewfster** [comment]
The bigger thing is that we have mostly moved on from 1.12 (I personally am not planning on touching 1.12 again unless enderio accidentally breaks their conduit API again and I have to release an update to gas conduits). At some point with someone's help and misc testing we did determine that removing [this line](https://github.com/Coders-After-Dark/Reskillable/blob/develop/1.12.2/src/main/java/codersafterdark/reskillable/client/base/ClientProxy.java#L42) does remove the crash. But at the time (and even more so now), I did/do not have the time to thoroughly test enough that it doesn't cause anything to break on our end, as git blame originally has that line being added to "fix" a few bugs on/with multiplayer. Personally I don't believe that line is needed, but as it was added quite a bit before I joined the team and Reskillable's code base has changed a decent bit since then, I did not want to remove it without a decent bit of testing.

It has been a while since I looked at how the advancement manager works but if I remember correctly I am fairly sure that we are **_not_** removing any other mod's advancements. (If I remember what the person I asked (who had added that line initially) the goal of that line is to force load the advancements on the client.)

**Comment by candys123** [comment]
> The bigger thing is that we have mostly moved on from 1.12 (I personally am not planning on touching 1.12 again unless enderio accidentally breaks their conduit API again and I have to release an update to gas conduits). At some point with someone's help and misc testing we did determine that removing [this line](https://github.com/Coders-After-Dark/Reskillable/blob/develop/1.12.2/src/main/java/codersafterdark/reskillable/client/base/ClientProxy.java#L42) does remove the crash. But at the time (and even more so now), I did/do not have the time to thoroughly test enough that it doesn't cause anything to break on our end, as git blame originally has that line being added to "fix" a few bugs on/with multiplayer. Personally I don't believe that line is needed, but as it was added quite a bit before I joined the team and Reskillable's code base has changed a decent bit since then, I did not want to remove it without a decent bit of testing.
> 
> It has been a while since I looked at how the advancement manager works but if I remember correctly I am fairly sure that we are **_not_** removing any other mod's advancements. (If I remember what the person I asked (who had added that line initially) the goal of that line is to force load the advancements on the client.)



Hi, 
sorry, but I seem to be stupid, would it be possible to send me this edited file? 
Everything I seem to try wont work. First I tried directly editing the class file, which didnt work, it seemed i needed to decompile it and edit and compile it again, however it didnt work. Afterwards I downloaded the github dev. forder, the one in your link. than i tried to compile the editet .java file, didnt work either. I tried now over 2 hours, is there any easy solution which makes it possible for me to edit this class file by myself?

Thank you.
